File: app/src/main/java/com/solutions/alphil/zambiajobalerts/classes/JobDao.java
package com.solutions.alphil.zambiajobalerts.classes;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

@Dao
public interface JobDao {
    @Query("SELECT * FROM jobs ORDER BY date DESC")
    LiveData<List<JobEntity>> getAllJobs();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertJobs(List<JobEntity> jobs);

    @Query("DELETE FROM jobs")
    void deleteAllJobs();

    @Query("SELECT * FROM jobs WHERE id = :jobId")
    JobEntity getJobById(int jobId);
    
    @Query("SELECT * FROM jobs WHERE title LIKE '%' || :query || '%' OR company LIKE '%' || :query || '%'")
    LiveData<List<JobEntity>> searchJobs(String query);
}

File: app/src/main/java/com/solutions/alphil/zambiajobalerts/classes/JobDao.kt
package com.solutions.alphil.zambiajobalerts.classes

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

@Dao
interface JobDao {
    @Query("SELECT * FROM jobs ORDER BY date DESC")
    fun getAllJobs(): LiveData<List<JobEntity>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertJobs(jobs: List<JobEntity>)

    @Query("DELETE FROM jobs")
    fun deleteAllJobs()

    @Query("SELECT * FROM jobs WHERE id = :jobId")
    fun getJobById(jobId: Int): JobEntity?

    @Query("SELECT * FROM jobs WHERE title LIKE '%' || :query || '%' OR company LIKE '%' || :query || '%'")
    fun searchJobs(query: String): LiveData<List<JobEntity>>
}
